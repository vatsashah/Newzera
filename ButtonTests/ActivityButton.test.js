// Importing required library
import React from 'react'
import { shallow, mount } from 'enzyme'

// Importing component to be tested
import ActionBubble from '../../../Components/Buttons/actionBubble'

describe('Activity Button should render without any error', () => {
  it('Activity Button should render correctly', () => {
    const component = shallow(<ActionBubble />)
    expect(component).toMatchSnapshot()
  })
  it('Activity Button of type share should render correctly', () => {
    const wrapper = mount(<ActionBubble activityType="Share" />).props()
    expect(wrapper.activityType).toEqual('Share')
  })
  it('Activity Button of type react should render correctly', () => {
    const wrapper = mount(<ActionBubble activityType="React" />).props()
    expect(wrapper.activityType).toBe('React')
  })
  it('Activity Button of type raven should render correctly', () => {
    const wrapper = mount(<ActionBubble activityType="Raven" />).props()
    expect(wrapper.activityType).toBe('Raven')
  })
})
