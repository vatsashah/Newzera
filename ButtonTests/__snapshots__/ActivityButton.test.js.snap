// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Button should render without any error Activity Button should render correctly 1`] = `
<div
  className="activityButtonBorder"
>
  <img
    alt="ActionButton"
    className="activityButton"
    src="Share.png"
  />
</div>
`;
